docs/topics/downloader-middleware.rst

```diff
@@ -922,13 +922,68 @@ RobotsTxtMiddleware
     To make sure Scrapy respects robots.txt make sure the middleware is enabled
     and the :setting:`ROBOTSTXT_OBEY` setting is enabled.
 
+    This middleware has to be combined with a `robots.txt <http://www.robotstxt.org/>`_ parser.
+
+    Scrapy ships with support for three `robots.txt <http://www.robotstxt.org/>`_ parsers:
+
+    * `RobotFileParser`_
+    * `Reppy`_
+    * `Robotexclusionrulesparser`_
+
+    You can change the `robots.txt <http://www.robotstxt.org/>`_ parser with the :setting:`ROBOTSTXT_PARSER`
+    setting. Or you can also :ref:`implement support for a new parser <support-for-new-robots-parser>`.
+
 .. reqmeta:: dont_obey_robotstxt
 
 If :attr:`Request.meta <scrapy.http.Request.meta>` has
 ``dont_obey_robotstxt`` key set to True
 the request will be ignored by this middleware even if
 :setting:`ROBOTSTXT_OBEY` is enabled.
 
+.. _support-for-new-robots-parser:
+
+Implementing support for a new parser
+-------------------------------------
+
+You can implement support for a new `robots.txt <http://www.robotstxt.org/>`_ parser by implementing the
+interface described below.
+
+.. module:: scrapy.extensions.robotstxtparser
+
+.. class:: RobotsParser
```

Comment:
:lipstick: You should be able to indicate the complete class path, thus making the `.. module` line above unnecessary.

```suggestion
.. class:: scrapy.extensions.robotstxtparser.RobotsParser
```